packages/modsecurity/data_stream/auditlog/elasticsearch/ingest_pipeline/apache-modsec.yml

```
@@ -0,0 +1,112 @@
+---
+description: Pipeline for apache modsecurity audit log.
+processors:
+  - rename:
+      field: json.transaction.time
+      target_field: _temps.date
+      ignore_missing: true
+  - date:
+      field: _temps.date
+      formats:
+         - d/MMM/yyyy:HH:mm:ss Z
+# rename ecs
+  - rename:
+      field: json.transaction.remote_address
+      target_field: source.ip
+      ignore_missing: true
+  - rename:
+      field: json.transaction.remote_port
+      target_field: source.port
+      ignore_missing: true
+  - grok:
+      field: json.request.request_line
+      patterns:
+        - "%{NOTSPACE:http.request.method} %{URIPATHPARAM:url.original}(?: HTTP/%{NUMBER:http.version})"
+  - set:
+      field: _temps.url
+      if: ctx.json.transaction.local_port == 443
+      value: "https://{{json.request.headers.Host}}:{{json.transaction.#local_port}}{{url.original}}"
+  - set:
+      field: _temps.url
+      if: ctx.json.transaction.local_port == 80
+      value: "http://{{json.request.headers.Host}}:{{json.transaction.#local_port}}{{url.original}}"
+  - uri_parts:
+      field: _temps.url
+      ignore_failure: true
+      keep_original: true
+      remove_if_successful: true
+  - rename:
+      field: json.response.status
+      target_field: http.response.status_code
+      ignore_missing: true
+  - rename:
+      field: json.response.headers.Content-Type
+      target_field: http.response.mime_type
+      ignore_missing: true
+  - rename:
+      field: json.response.headers.Content-Length
+      target_field: http.response.bytes
+      ignore_missing: true
+  - convert:
+      field: http.response.bytes
+      type: long
+  - rename:
+      field: json.audit_data.messages
+      target_field: modsec.audit.messages
+      ignore_missing: true
+  - foreach:
+      field: modsec.audit.messages
+      ignore_missing: true
+      processor:
+        grok:
+          field: '_ingest._value'
+          ignore_failure: true
+          patterns:
+             - '%{GREEDYDATA} \[file "%{GREEDYDATA:modsec.audit.details.file}"\] \[line "%{GREEDYDATA:_temps.line_number}"\] \[id "%{GREEDYDATA:modsec.audit.details.rule_id}"\] \[msg "%{GREEDYDATA:modsec.audit.details.msg}"\] \[data "%{GREEDYDATA:modsec.audit.details.match}"\] \[severity "%{GREEDYDATA:modsec.audit.details.severity}"\] \[ver "%{DATA:modsec.audit.details.version}"\] %{GREEDYDATA:_temps.tags}'
+  - lowercase:
+      field: modsec.audit.details.severity
+      ignore_missing: true
```

This would be nice to have, but the options available to make this work and to have it provide a benefit are not here. So I think that for this change it should be deleted with a view to revisiting it in the future.

```suggestion
  - rename:
      field: json.audit_data.messages
      target_field: modsec.audit.details
      ignore_missing: true
  - script:
      lang: painless
      ignore_failure: true
      source: |
        if (ctx.modsec?.audit?.details == null || ctx.modsec.audit.details.length == 0) {
          return;
        }
        def details = ctx.modsec.audit.details;
        def messages = new ArrayList();
        for (def i = 0; i < details.length; i++) {
          def idx = details[i].indexOf(' ['); // Find first key value mark.
          if (idx < 0) {
            idx = details[i].length;
          }
          messages.add(details[i].substring(0, idx));
        }
        ctx.modsec.audit.messages = messages

```